### PR TITLE
Enhance `LogsBloomFilter` to always pad target to 256 bytes

### PR DESCRIPTION
**Pull request by bilyana-gospodinova**
**Description**:
Allow record-file logsBloom aggregation when the file has no top-level contract results, including hook-driven synthetic logs.

* Change `LogsBloomFilter.or` to treat null and empty blooms as absent and to pad a short or missing target to 256 bytes instead of throwing.
* Initialize the file bloom to 256 bytes in `SyntheticLogListener` before OR-ing contract results linked via `getContractRelatedParent()`.
* Add unit tests for `null`, empty, and 256-byte `or` combinations.
* Add parser tests for a HAPI 0.70 hook-driven transfer and for a multi-file batch whose last file has no top-level contract transactions.

Ref: MN-140

### CODE REVIEW

File: common/src/main/java/org/hiero/mirror/common/util/LogsBloomFilter.java [hunk 1]
```diff
@@ -48,27 +49,26 @@ public boolean couldContain(final byte[] bloom) {
     }
 
     /**
-     * Utility method that mutates the target array by aggregating its content with a passed source that
-     * should have a matching length
-     * */
+     * Utility method that mutates the target array by aggregating its content with a passed source.
+     * A null, empty, or shorter target is replaced by a new 256-byte buffer. Absent blooms (null or
```
Comment (IvanKavaldzhiev) [inline common/src/main/java/org/hiero/mirror/common/util/LogsBloomFilter.java:53]: Do we really need to allow shorter targets? I think these should be treated as incorrect data and that's why we threw an exception. Otherwise we are silently masking an issue from the record/block stream.

File: common/src/main/java/org/hiero/mirror/common/util/LogsBloomFilter.java [hunk 2]
```diff
@@ -48,27 +49,26 @@ public boolean couldContain(final byte[] bloom) {
...
-        }
+        final var destination = target != null && target.length == BYTE_SIZE
+                ? target
+                : ArrayUtils.isEmpty(target) ? new byte[BYTE_SIZE] : Arrays.copyOf(target, BYTE_SIZE);
 
-        if (target.length == BYTE_SIZE) {
-            return target;
-        } else {
-            return new byte[BYTE_SIZE];
+        for (int i = 0, n = Math.min(source.length, destination.length); i < n; i++) {
```
Comment (IvanKavaldzhiev) [inline common/src/main/java/org/hiero/mirror/common/util/LogsBloomFilter.java:68]: Math.min can also hide a corrupted or wrong data from the stream.

File: importer/src/test/java/org/hiero/mirror/importer/parser/record/RecordFileParserIntegrationTest.java [hunk 3]
```diff
@@ -349,6 +362,46 @@ void topicMessage() {
...
+                .recordItem(TransactionType.CRYPTOTRANSFER)
+                .build();
+
+        recordFileParser.parse(List.of(firstFile, lastFile));
+
+        assertThat(recordFileRepository.findAll()).hasSize(2);
+        assertThat(contractLogRepository.findAll()).isNotEmpty();
+        final var lastPersisted =
+                recordFileRepository.findById(lastFile.getConsensusEnd()).orElseThrow();
+        assertThat(lastPersisted.getLogsBloom()).hasSize(LogsBloomFilter.BYTE_SIZE);
```
Comment (IvanKavaldzhiev) [inline importer/src/test/java/org/hiero/mirror/importer/parser/record/RecordFileParserIntegrationTest.java:402]: If the CRYPTO_TRANSFER is not related to a token transfer, should it have a non-null logs bloom?

**Review by IvanKavaldzhiev — APPROVED** [review]
LGTM

**Review by steven-sheehy — APPROVED** [review]
